pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -276,7 +276,7 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 			err = og.verifyVolumeIsSafeToDetach(volumeToDetach)
 		}
 		if err == nil {
-			err = volumeDetacher.Detach(volumeName, volumeToDetach.NodeName)
+			err = volumeDetacher.Detach(volumeToDetach.VolumeSpec, volumeName, volumeToDetach.NodeName)
```

Comment:
I am out of touch with the volume code these days, but once upon a time it was impossible to guarantee that the spec was available at Detach time, and therefore the name had to carry enough information to do a detach without the spec.

In fact, this is a large part of the reason that Attacher and Detacher exist.

Has ground truth changed such that I *know* the spec is available at detach time?